lib/analysis/modules/ies_enkf.c

```diff
@@ -0,0 +1,1059 @@
+                       matrix_type * Yin ,    // Ensemble of predicted measurements
+                       matrix_type * Rin ,    // Measurement error covariance matrix (not used)
+                       matrix_type * dObs ,   // Actual observations (not used)
+                       matrix_type * Ein ,    // Ensemble of observation perturbations
+                       matrix_type * Din ,    // (d+E-Y) Ensemble of perturbed observations - Y
+                       const module_info_type * module_info,
+                       rng_type * rng) {
+
+   ies_enkf_data_type * data = ies_enkf_data_safe_cast( module_data );
+
+   int nrobs_msk     =bool_vector_size( data->obs_mask ); // Total number of observations
+   int nrobs_inp     =matrix_get_rows( Yin );             // Number of active observations input in current iteration
+   int nrobs         =nrobs_inp;                          // Number of selected active observations
+
+
+   int ens_size_msk  =bool_vector_size(data->ens_mask);   // Total number of realizations
+   int ens_size      =matrix_get_columns( Yin );          // Number of active realizations in current iteration
+
+   int state_size    = matrix_get_rows( A );
+
+   double rcond;
+   int nrsing=0;
+
+   if (data->state_size == 0) data->state_size=state_size;
+
+
+   data->iteration_nr         = data->iteration_nr+1 ;
+   data->ies_steplength       = ies_enkf_config_get_ies_steplength( data->config );
+   data->ies_subspace         = ies_enkf_config_get_ies_subspace( data->config );
+   data->ies_inversion        = ies_enkf_config_get_ies_inversion( data->config );
+   data->ies_logfile          = ies_enkf_config_get_ies_logfile( data->config );
+   data->ies_debug            = ies_enkf_config_get_ies_debug( data->config );
+
+   data->truncation           = ies_enkf_config_get_truncation( data->config );
+   data->subspace_dimension   = ies_enkf_config_get_enkf_subspace_dimension( data->config );
+
+   data->gauss_newton_conv    = ies_enkf_config_get_gauss_newton_conv( data->config );
+
+   if (data->iteration_nr > 3) data->ies_steplength=data->ies_steplength/2;
+   if (data->iteration_nr > 6) data->ies_steplength=data->ies_steplength/4;
+
+
+   FILE *fp;
+   if (data->iteration_nr == 1){
+      fp=freopen(data->ies_logfile, "w", stdout);
+   } else {
+      fp=freopen(data->ies_logfile, "a", stdout);
+   }
+
+   fprintf(stdout,"\n\n\n***********************************************************************\n");
+   fprintf(stdout,"IES Iteration   = %d\n", data->iteration_nr);
+   fprintf(stdout,"----ies_steplength  = %f\n", data->ies_steplength);
+   fprintf(stdout,"----ies_inversion   = %d\n", data->ies_inversion);
+   fprintf(stdout,"----ies_debug       = %d\n", data->ies_debug);
+   fprintf(stdout,"----truncation      = %f %d\n", data->truncation, data->subspace_dimension);
+   fprintf(stdout,"----ies_logfile     = %s\n", data->ies_logfile);
+   bool dbg = ies_enkf_config_get_ies_debug( data->config ) ;
```

Comment:
Maybe instead of a `bool` for debug it could be a log_level: https://github.com/Statoil/libres/blob/master/lib/include/ert/res_util/log.hpp#L31- then you can do 
```
message_level_type log_level = ies_config_get_ies_log_level( data->config );
if (log_level <= LOG_DEBUG)
     // This is debug message which only comes when you have set the config log level to DEBUG (the lowest value).
     fprintf( ... );

if (log_level <= LOG_CRITICAL)
   // Complete f...up - this should always come ...
```

That way all the nice debug messages you have configured can be retained from some time?